File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let ans = 0;
    function getAnswer(idx, sum) {
        if(idx <numbers.length) {
            getAnswer(idx+1, sum + numbers[idx])
            getAnswer(idx+1, sum - numbers[idx])
        } else {
            if(sum === target) ans++;
            return;
        }
    }
    getAnswer(0,0)
    return ans;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    cnt = 0
    def getAnswer(idx, summary):
        nonlocal cnt
        if idx == len(numbers):
            if summary == target: cnt += 1
            return
        else:
            getAnswer(idx+1,summary + numbers[idx])
            getAnswer(idx+1,summary - numbers[idx])
            
    getAnswer(0,0)
    return cnt

